java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -33,7 +33,12 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 
-typealias ParticleConstructor = suspend () -> Particle
+sealed class ParticleConstructor {
+    // TODO(alxr): Do these need to be suspend functions? If so, I have a solution to achieve that.
+    abstract class Empty : () -> Particle, ParticleConstructor()
+    abstract class Spec : (Plan.Particle) -> Particle, ParticleConstructor()
```

Comment:
Don't repeat the wrong naming - Plan.Particle is not a spec. Maybe s/Spec/Reflective/ - to indicate that the particle is using a kind of reflection on the way it is instantiated.

Reply:
I agree `Spec` isn't good. I think `Reflective` is still not quite right. I'll give the name some thought.